x/mongo/driver/operation/count.go

```diff
@@ -54,15 +56,38 @@ func buildCountResult(response bsoncore.Document, srvr driver.Server) (CountResu
 	cr := CountResult{}
 	for _, element := range elements {
 		switch element.Key() {
-		case "n":
+		case "n": // for count using original command
 			var ok bool
 			cr.N, ok = element.Value().AsInt64OK()
 			if !ok {
-				err = fmt.Errorf("response field 'n' is type int64, but received BSON type %s", element.Value().Type)
```

Comment:
Why did this change? We know that the element's key is `n` in this case, so why do we need the error message to be generic?